### ISSUE TITLE
Buffer overflow when connecting

### ISSUE THREAD

**Issue by MathyV** [adrienverge/openfortivpn#51]
I am running into a buffer overflow when connecting to a (out of my control) VPN server. I traced it down to the function parse_config which does not properly check the boundaries when trying to read the value of the "text6" input field.

For the server I am connecting to this field is empty so naturally the search for '/' picks up the next closing html tag which results in the buffer overflow when copying the data.

This is what is returned by the server for /remote/fortisslvpn:

```html
<HTML>

<HEAD>

<META http-equiv="Content-Type" content="text/html; charset=UTF-8">
<META http-equiv="Pragma" content="no-cache">
<META http-equiv="cache-control" content="no-cache">
<META http-equiv="cache-control" content="must-revalidate">
<link href="/sslvpn/css/sslvpn.css" rel="stylesheet" type="text/css">

<TITLE>Fortinet SSL-VPN Client Version 1.0</TITLE>

<!-- SSL-VPN protocol version:
embed.FGTversion = 1;
fortisslvpn.FGTversion = 1
-->

</HEAD>
<BODY class=main>
<form name="fortisslvpn">
<input type="hidden" NAME="text6" VALUE="">
<input type="hidden" NAME="text3" value="server.name.is.here:443">
<input type="hidden" NAME="text7" value="0">
```

It is possible to connect using forticlient. Let me know if I can provide more information. I do not have access to or more information on the server.

**Comment by adrienverge** [comment]
Hi @MathyV, thanks for finding this and tracking it down to the `"text6"` parsing!

Can you tell me if #52 solves the problem?

**Comment by MathyV** [comment]
The buffer overflow is fixed with #52 and I can now log into the VPN, awesome!

**Comment by adrienverge** [comment]
:+1:
